crates/core/app/tests/mock_consensus_staking.rs

```diff
@@ -101,29 +101,23 @@ async fn mock_consensus_can_delegate_to_a_validator() -> anyhow::Result<()> {
 
     // Make a transaction that defines a new validator.
     let new_validator_vk = node.keyring_mut().add_key();
-    let (new_validator_pb, _) = self::common::generate_penumbra_validator(&new_validator_vk);
+    let (new_validator_pb, _, new_validator_spend_key) =
+        self::common::generate_penumbra_validator(&new_validator_vk);
     let new_validator = penumbra_stake::validator::Validator::try_from(new_validator_pb.clone())?;
-    let tx = {
+    let mut plan = {
         use {
-            penumbra_keys::keys::{SpendKey, SpendKeyBytes},
             penumbra_proto::Message,
             penumbra_stake::validator,
             penumbra_transaction::{ActionPlan, TransactionParameters, TransactionPlan},
-            rand::Rng,
             rand_core::OsRng,
         };
-        let auth_sig = {
-            // TODO(kate): for now, we generate a throwaway key. figure out how to present a
-            // *valid* authentication signature here later.
-            let bytes = new_validator_pb.encode_to_vec();
-            let seed = SpendKeyBytes(OsRng.gen());
-            SpendKey::from(seed).spend_auth_key().sign(OsRng, &bytes)
-        };
+        let bytes = new_validator_pb.encode_to_vec();
+        let auth_sig = new_validator_spend_key.spend_auth_key().sign(OsRng, &bytes);
         let action = ActionPlan::ValidatorDefinition(validator::Definition {
-            validator,
+            validator: new_validator.clone(),
             auth_sig,
         });
```

Comment:
:point_up: right here, we need the spend auth key to sign the validator definition

Reply:
ceb5d62